3-reveal/migrations/5-IRS/2-Zambia-2019/deploy/process_structure_geo_hierarchy_queue.psql

```diff
@@ -8,10 +8,52 @@ BEGIN;
 -- :"schema" variable - and default to public.
 SET search_path TO :"schema",public;
 
+CREATE FUNCTION process_structure_geo_hierarchy_full() RETURNS VOID AS $$
+-- this function is used when we need to reprocess all locations
+-- WARNING: this is an expensive query!!
+BEGIN
+    -- first drop structure_geo_hierarchy_full
+    DROP MATERIALIZED VIEW IF EXISTS structure_geo_hierarchy_full;
+    -- then recreate it
+    CREATE MATERIALIZED VIEW structure_geo_hierarchy_full AS
+    SELECT
+            DISTINCT ON (locations.id) locations.id AS id,
+            jurisdictions.id AS jurisdiction_id,
+            jurisdictions.geographic_level AS jurisdiction_depth,
+            geo_jurisdictions.id AS geo_jurisdiction_id,
+            geo_jurisdictions.geographic_level AS geo_jurisdiction_depth,
+            public.ST_WITHIN(locations.geometry, geo_jurisdictions.geometry) AS geo_strict_within
+        FROM locations
+        LEFT JOIN jurisdictions
+            AS geo_jurisdictions
+            ON public.ST_DWITHIN(locations.geometry, geo_jurisdictions.geometry, 150.0 / 111111.0)
```

Comment:
this is the expensive part? how did we come up w/the distance metric of 150/etc

Reply:
can we set `use_spheroid` the final arg to false? docs say this is faster

https://postgis.net/docs/ST_DWithin.html

Reply:
We got the 150m from the client.  i.e. if a structure is outside of a 150m buffer from a jurisdiction's boundary then it does not belong to that jurisdiction

Reply:
oh I see so b/c these are geometries and not geographies we do that 111111 correction

hmm, is it possible to use geographies?

Reply:
ok nevermind this doesn't make sense, since these are geometries they default to NOT use the sphere

reading more about this, the deal w/geometries is they use an ellipsoidal model so the calculations are more accurate, b/c they take the curvature of the earth into account, but if we're using the less accurate geometries anyways this isn't relevant as long as we're using 2D geometries

it sounds like it would be more appropriate for us to use geographies, and set the spheroid arg to false here at some point (since our data is geographic) but this is a separate chunk of probably pretty big work we can not worry